### PR TITLE
Components

### PR DESCRIPTION
**Pull request by Motoblock**
1. Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module01
2. Screenshot:
image
3. Deploy: https://rs-components.netlify.app/
4. Done 13-03-2023 / deadline 20-03-2023
5. Score: 15 / 15

### CODE REVIEW

File: src/page/main/main.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <>
+        <h2>Paws</h2>
+        <MySearch {...this.prop} />
+        <Cards />
+      </>
+    );
+  }
+}
+
+function Cards() {
```
Comment (kuzanatoliy) [inline src/page/main/main.tsx]:
```suggestion
const CardList = (({ catsData })) => <div>...</div>
```
It is possible to use shorter construction here

File: src/page/main/main.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import MySearch from './../../component/UI/input/MySearch';
+import catsData from './../../assets/data/cats';
+import Card from './../../component/UI/card/Card';
+import './../main/main.module.css';
+import SearchProps from './../../component/UI/input/interface';
+
+class Main extends React.Component {
```
Comment (kuzanatoliy) [inline src/page/main/main.tsx]: Please use rule one file one component

File: src/page/main/main.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <div className="cards__box">
+      {catsData.map((el) => (
+        <Card key={el.id} {...el} />
+      ))}
+    </div>
+  );
+}
+
+export default Main;
```
Comment (kuzanatoliy) [inline src/page/main/main.tsx]: Please use named exports instead of default

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true,
+    "baseUrl": "./src",
+    "resolveJsonModule": true
+  },
+  "include": ["./src/**/*", "styled.d.ts"]
+}
```
Comment (kuzanatoliy) [inline tsconfig.json]: Empty line in end of files

File: src/page/about/about.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+function About() {
+  return (
+    <>
+      <h2>О котиках</h2>
+      <br />
+      <p>
+        Сейчас в мире живет свыше 500 миллионов домашних кошек. Самая «кошачья» страна — это
```
Comment (kuzanatoliy) [inline src/page/about/about.tsx]: Please try to use English

File: src/component/UI/card/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  colorOption: string;
+  age: number;
+  raiting: number;
+  counts: number;
+  catterys: string;
+  description: string;
+  image: string;
+}
+class Card extends React.Component<CardCatProps, { value: string }> {
+  // constructor(props: CardCatProps) {
```
Comment (kuzanatoliy) [inline src/component/UI/card/Card.tsx]: Looks like you forget to remove it

File: src/component/UI/card/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import stCard from './../card/card.module.css';
+
+export interface CardCatProps {
```
Comment (kuzanatoliy) [inline src/component/UI/card/Card.tsx]:
```suggestion
export interface ICardCatProps {
```

File: src/component/Layout.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Outlet } from 'react-router-dom';
+import Footer from './Footer';
+import Header from './Header';
+
+const Layout = () => {
+  return (
+    <>
```
Comment (kuzanatoliy) [inline src/component/Layout.tsx]: The fragment doesn't make sense here

File: src/component/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+class Header extends Component<Props, { title: string }> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      title: '',
+    };
+  }
+
+  componentDidMount() {
+    if (location.pathname === '/') this.setState({ title: 'Home page' });
```
Comment (kuzanatoliy) [inline src/component/Header.tsx]: Logically switch would have more sense here

**Review by kuzanatoliy — COMMENTED** [review]
Please use preprocessors for styles https://vitejs.dev/guide/features.html#css

File: src/component/UI/input/input.test.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
Comment (kuzanatoliy) [inline src/component/UI/input/input.test.tsx:1]: Please use one name standard for components

File: src/page/about/about.tsx (hunk 5 above)
Comment (Motoblock) [reply src/page/about/about.tsx]: yes, well

File: src/page/main/main.tsx (hunk 1 above)
Comment (Motoblock) [reply src/page/main/main.tsx]: yes, well

File: tsconfig.json (hunk 4 above)
Comment (Motoblock) [reply tsconfig.json]: add

File: src/component/UI/card/Card.tsx (hunk 7 above)
Comment (Motoblock) [reply src/component/UI/card/Card.tsx]: oops )

File: src/component/UI/card/Card.tsx (hunk 6 above)
Comment (Motoblock) [reply src/component/UI/card/Card.tsx]: oops...

File: src/component/Layout.tsx (hunk 8 above)
Comment (Motoblock) [reply src/component/Layout.tsx]: yes, not needed

File: src/component/Header.tsx (hunk 9 above)
Comment (Motoblock) [reply src/component/Header.tsx]: yes, logically

File: src/page/main/main.tsx (hunk 3 above)
Comment (Motoblock) [reply src/page/main/main.tsx]: I understand correctly that then this is how to do for all classes/functions?
export class Main extends React.Component {...

File: src/component/UI/input/input.test.tsx (hunk 10 above)
Comment (Motoblock) [reply src/component/UI/input/input.test.tsx:1]: yes, well

File: src/page/main/main.tsx (hunk 2 above)
Comment (Motoblock) [reply src/page/main/main.tsx]: yes, well

**Pull request merged**